pkg/kubectl/cmd/expose.go

```diff
@@ -133,7 +134,11 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	}
 	names := generator.ParamNames()
 	params := kubectl.MakeParams(cmd, names)
-	params["default-name"] = info.Name
+	name := info.Name
+	if len(name) > validation.DNS952LabelMaxLength {
+		name = name[:validation.DNS952LabelMaxLength]
```

Comment:
We should only truncate a default name, not a user specified name (if you explicitly set one that is too long we should fail).  Can you add that logic and a distinct test case for it?